src/lib.rs

```diff
@@ -164,9 +164,10 @@ pub type Result<T> = ::std::result::Result<T, Error>;
 ///
 /// Results are returned in depth first fashion, with directories yielded
 /// before their contents. If [`contents_first`] is true, contents are yielded
-/// before their directories. The order is unspecified but if [`sort_by`] is
-/// given, directory entries are sorted according to this function. Directory
-/// entries `.` and `..` are always omitted.
+/// before their directories. If [`sort_by`] the directory entries are sorted
+/// according to this function. If it is not then the order is unspecified
+/// (see [`sort_by`] for more information. Directory entries `.` and `..` are
+/// always omitted.
```

Comment:
Could you clean this up a bit please? I think you're missing a word or two  in `If [sort_by] the`, and I don't see a closing parenthesis for the new note added.